cdap-docs/users-manual/source/etl/creating.rst

```diff
@@ -0,0 +1,422 @@
+                      }
+             }
+  }
+
+Configuring ETL Adapters and Plugins
+====================================
+
+.. highlight:: console
+
+In order to configure an ETL Adapter, you’ll need information about the different ETL
+plugins you are using: which sources, transforms, and sinks are available, and what
+properties need to be specified for each of them.
+
+This information can be retrieved from the platform using the CDAP ETL Adapter HTTP
+RESTful API. You can access this with the CDAP CLI (Command Line Interface), ``curl``
+calls on a command line, or [through the CDAP UI].
+
+In each of the following endpoints, ``<template-id>`` refers to either ``etlBatch`` or
+``etlRealtime``. For example, if you fetch the information on sources for the
+``etlRealtime`` template, it will provide information about all the realtime sources that
+are available in the platform.
+
+If you have added your own Application Templates [link] or Plugins [link], then in the endpoints
+described below, you can substitute your Template IDs and Types as appropriate. 
+
+These endpoints are specific to ETL Application Templates and their Plugins. Endpoints
+that are applicable to all plugins [link] (including the ones you might have created) are
+documented in the CDAP Reference Manual, HTTP RESTful API [link].
+
+Retrieving Available Plugin Types
+---------------------------------
+
+To retrieve the available Plugin types (extensions) for an Application Template, use::
+
+  GET <base-url>/templates/<template-id>
+  
+  
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``<template-id>``
+     - Template ID, one of either ``etlBatch`` or ``etlRealtime``
+
+
+Retrieving Available Plugins
+----------------------------
+
+To retrieve the available Plugins of a given type (extension) for an Application Template, use::
+
+  GET <base-url>/templates/<template-id>/extensions/<type>
+  
+  
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``<template-id>``
+     - Template ID, one of either ``etlBatch`` or ``etlRealtime``
+   * - ``<type>``
+     - Plugin type, one of either ``source``, ``transform``, or ``sink``
+
+
+Retrieving Plugin Details
+-------------------------
+
+To retrieve the Plugin details for an Application Template, use::
+
+  GET <base-url>/templates/<template-id>/extensions/<type>/plugins/<plugin-id>
+  
+  
+.. list-table::
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Parameter
+     - Description
+   * - ``<template-id>``
+     - Template ID, one of either ``etlBatch`` or ``etlRealtime``
+   * - ``<type>``
+     - Plugin type, one of either ``source``, ``transform``, or ``sink``
+   * - ``<plugin-id>``
+     - Plugin ID (name), as obtained from a previous call to retrieve the available plugins
+
+
+Examples
+--------
+
+::
+
+  # Batch
+
+  # Retrieving the available types of plugins of the ETL Batch Application Template
```

Comment:
I think we should format the output through " python -m json.tool " to get better formatting.